forms-flow-components/src/components/CustomComponents/FormBuilderModal.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useRef, useEffect } from "react";
+import Modal from "react-bootstrap/Modal";
+import { CustomButton } from "./Button";
+import { FormTextArea } from "./FormTextArea";
+import { CloseIcon } from "../SvgIcons/index";
+import { FormInput } from "./FormInput";
+
+interface BuildFormModalProps {
+  showBuildForm: boolean;
+  onClose?: () => void;
+  handleChange?: (
+    field: string,
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => void;
+  primaryBtnAction?: () => void;
+  secondaryBtnAction?: () => void;
+  setFormDescription?: (value: string) => void;
+  setNameError?: (value: string) => void;
+  nameError?: string;
+  formSubmitted?: boolean;
+  modalHeader?: string;
+  nameLabel?: string;
+  descriptionLabel?: string;
+  primaryBtnLabel?: string;
+  secondaryBtnLabel?: string;
+  nameValidationOnBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
+  primaryBtndataTestid?: string;
+  secondoryBtndataTestid?: string;
+  primaryBtnariaLabel?: string;
+  secondoryBtnariaLabel?: string;
+  closedataTestid?: string;
+  nameInputDataTestid?: string;
+  descriptionDataTestid?: string;
+  placeholderForForm?: string;
+  placeholderForDescription?: string;
+}
+
+export const FormBuilderModal: React.FC<BuildFormModalProps> = React.memo(
+  ({
+    showBuildForm,
+    onClose,
+    handleChange,
+    primaryBtnAction,
+    secondaryBtnAction = onClose, // Default to onClose if not provided
+    setFormDescription,
+    setNameError,
+    nameError,
+    formSubmitted,
+    modalHeader,
+    nameLabel = "Name",
+    descriptionLabel = "Form Description",
+    primaryBtnLabel = "Save and Edit form",
+    secondaryBtnLabel = "Cancel",
+    nameValidationOnBlur,
+    primaryBtndataTestid = "confirm-button",
+    secondoryBtndataTestid = "cancel-button",
+    primaryBtnariaLabel = "Create and Edit form",
+    secondoryBtnariaLabel = "Cancel",
+    closedataTestid = "close",
+    nameInputDataTestid = "form-name",
+    descriptionDataTestid = "form-description",
+    placeholderForForm = "Form",
+    placeholderForDescription = "Form Description",
```

Comment:
We dont need default placeholder. its not part of design . Please remove default placeholder